pokedex/src/components/Detail.js

```diff
@@ -1,176 +1,77 @@
 
 const Detail = () => {
-  const [data, setData] = React.useState({});
-  const [isLoading, setIsLoading] = React.useState(true);
+  const [pokemonInfo, setPokemonInfo] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
   let { page_id } = useParams();
+  let navigate = useNavigate();
 
-  const backButtonClicked = () => {
-    window.history.back();
-  };
-
-  React.useEffect(() => {
+  useEffect(() => {
     fetch("https://intern-pokedex.myriadapps.com/api/v1/pokemon/" + page_id)
       .then((res) => res.json())
       .then((json) => {
-        setData(json.data);
+        setPokemonInfo(json.data);
         setIsLoading(false);
       });
-  }, []);
+  });
 
-  const getStyle = (hp) => {
-    const length = (hp / 250) * 93;
-    return length + "%";
-  };
-  
-  if (!isLoading) {
-    const { id, name, image, types, height, weight, abilities, egg_groups, stats, genus, description } = data;
-    const { hp, speed, attack, defense } = stats;
-    const specialAttack = stats["special-attack"];
-    const specialDefense = stats["special-defense"];
+  if (!parseInt(page_id) || page_id > 553) {
+    return <Invalid />;
+  } else if (!isLoading) {
+    const { id, name, image, types, height, weight, abilities, egg_groups, stats, genus, description } = pokemonInfo;
     return (
-      <div className="detail-page" id={types[0]}>
-        <header className="top">
-          <button onClick={backButtonClicked} className="arrow back">
+      <div className="fill-window" id={types[0]}>
+        <header className="top inline-center">
+          <button onClick={() => navigate(-1)} className="arrow back fixed">
             &#10132;
           </button>
-          <h1 className="page-title">{name}</h1>
+          <h1 className="page-title center detail-page-title">{name}</h1>
         </header>
-        <div className="detail-card">
-          <div className="card-head">
+        <div className="center detail-card">
+          <div className="flex-container">
             <div className="card-title">
-              <h1>{name}</h1>
-              <p> &nbsp; #{id}</p>
-            </div>
-            <div className="card-buttons">
-              {Object.keys(types).map((key) => (
-                <button className="card-button" id={types[key]} key={key}>
-                  <span>{types[key]}</span>
-                </button>
-              ))}
+              <h1>{name} </h1>
+              <h2 className="page-title"> &nbsp; #{id}</h2>
             </div>
+            <Types types={types} />
           </div>
           <hr />
-          <div className="container">
-            <div>
-              <img className="card-image" src={image} alt={name} />
+          <div className="flex-container">
+            <div className="fle">
+              <img
+                className="image detail-image"
+                src={"https://assets.pokemon.com/assets/cms2/img/pokedex/full/" + getIdWithZeros(id) + ".png"}
```

Comment:
A minor improvement here would be to use template literals:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals